client/src/reducers/categoricalSelectionState.js

```diff
@@ -0,0 +1,93 @@
+import _ from "lodash";
+
+import { ControlsHelpers } from "../util/stateManager";
+import * as globals from "../globals";
+
+function maxCategoryItems(state) {
+  return _.get(
+    state.config,
+    "parameters.max-category-items",
+    globals.configDefaults.parameters["max-category-items"]
+  );
+}
+
+const CategoricalSelectionState = (
+  state,
+  action,
+  nextSharedState,
+  prevSharedState
+) => {
+  switch (action.type) {
+    case "initial data load complete (universe exists)":
+    case "set World to current selection":
+    case "reset World to eq Universe": {
+      const { world } = nextSharedState;
+      return ControlsHelpers.createCategoricalSelectionState(
+        maxCategoryItems(prevSharedState),
+        world
+      );
+    }
+
+    case "categorical metadata filter select": {
+      const newCategorySelected = Array.from(
+        state[action.metadataField].categorySelected
+      );
+      newCategorySelected[action.categoryIndex] = true;
+      const newCategoricalSelectionState = {
+        ...state,
+        [action.metadataField]: {
+          ...state[action.metadataField],
+          categorySelected: newCategorySelected
+        }
+      };
+      return newCategoricalSelectionState;
+    }
+
+    case "categorical metadata filter deselect": {
+      const newCategorySelected = Array.from(
+        state[action.metadataField].categorySelected
+      );
+      newCategorySelected[action.categoryIndex] = false;
+      const newCategoricalSelectionState = {
+        ...state,
+        [action.metadataField]: {
+          ...state[action.metadataField],
+          categorySelected: newCategorySelected
+        }
+      };
+      return newCategoricalSelectionState;
+    }
+
+    case "categorical metadata filter none of these": {
+      const newCategoricalSelectionState = {
+        ...state,
+        [action.metadataField]: {
+          ...state[action.metadataField],
+          categorySelected: Array.from(
+            state[action.metadataField].categorySelected
+          ).fill(false)
+        }
+      };
+      return newCategoricalSelectionState;
+    }
+
+    case "categorical metadata filter all of these": {
+      const newCategoricalSelectionState = {
+        ...state,
+        [action.metadataField]: {
+          ...state[action.metadataField],
+          categorySelected: Array.from(
+            state[action.metadataField].categorySelected
+          ).fill(true)
```

Comment:
Suggest a comment on this line for posterity

Reply:
I'm not sure what you are proposing - can you elaborate?

Reply:
This is one of the 'chewier' lines, those reading from a non JS background might appreciate a plaintext reading of this operation. If you disagree feel free to skip.

Reply:
lol, you wrote this code :-)

I'll add comments.

Reply:
OK, added a comment about the intent (to all four places we do something similar).  LMK if that resolves it, or propose a comment if you want something different (I wasn't sure if "chewie" referred to the intent or the code).

Reply:
Those are perfect!